lib/customer.rb

```diff
@@ -0,0 +1,40 @@
+require 'csv'
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address
+
+  DEFAULT_CSV_FILE = "data/customers.csv"
+
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all(csv_file = DEFAULT_CSV_FILE)
+    all_customers = []
+
+    CSV.open(csv_file, "r").each do |customer|
+      all_customers << self.csv_to_customer(customer)
+    end
+
+    return all_customers #array of customer objects
+  end
+
+  def self.find(id, csv_file = DEFAULT_CSV_FILE)
+    CSV.open(csv_file, "r").each do |customer|
+      if customer[0].to_i == id
+        return self.csv_to_customer(customer)
```

Comment:
This method should call `.all` instead of repeating the logic to open the CSV file.